dice_ml/diverse_counterfactuals.py

```diff
@@ -6,7 +6,7 @@
 class CounterfactualExamples:
     """A class to store and visualize the resulting counterfactual explanations."""
 
-    def __init__(self, data_interface, test_instance, test_pred, final_cfs, final_cfs_preds, final_cfs_sparse=None, cfs_preds_sparse=None, posthoc_sparsity_param=0, desired_class="opposite"):
+    def __init__(self, data_interface, test_instance, test_pred, final_cfs, final_cfs_preds, final_cfs_sparse=None, cfs_preds_sparse=None, posthoc_sparsity_param=0, desired_class="opposite", encoding='one-hot'):
```

Comment:
need to think more about the encoding parameter. For a user, it can be a problem if their ML model uses one kind of encoding, and then DiCE forces on them to choose a different kind of encoding (e.g., onehotencoder from sklearn or get_dummies from pandas). 
The same issue can also come with labelencoder. It will be nice to support any encoder that a user provides. 
To do that, we can say that the encoding parameter can be either a string or a function. If it's a string, we apply the default encoding. If it is a function, then we directly apply that function to the input data.